activiti-core/activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/ProcessRuntimeImpl.java

```diff
@@ -491,4 +496,39 @@ public Deployment selectLatestDeployment(){
         );
     }
 
+    public org.activiti.engine.runtime.ProcessInstance internalProcessInstance(String processInstanceId) {
+        org.activiti.engine.runtime.ProcessInstance internalProcessInstance = runtimeService
+            .createProcessInstanceQuery()
+            .processInstanceId(processInstanceId)
+            .singleResult();
+
+        if (internalProcessInstance == null) {
+            throw new NotFoundException("Unable to find process instance for the given id:'" + processInstanceId + "'");
+        }
+        return internalProcessInstance;
+    }
+
+    private boolean canReadProcessInstance(org.activiti.engine.runtime.ProcessInstance processInstance) {
+        return securityPoliciesManager.canRead(processInstance.getProcessDefinitionKey()) &&
+            (securityManager.getAuthenticatedUserId().equals(processInstance.getStartUserId()) ||
+                isATaskAssigneeOrACandidate(processInstance.getProcessInstanceId()));
+    }
+
+    private boolean canWriteProcessInstance(org.activiti.engine.runtime.ProcessInstance processInstance) {
+        return securityPoliciesManager.canWrite(processInstance.getProcessDefinitionKey()) &&
+            securityManager.getAuthenticatedUserId().equals(processInstance.getStartUserId());
+    }
+
+    private boolean isATaskAssigneeOrACandidate(String processInstanceId) {
+        String authenticatedUserId = securityManager.getAuthenticatedUserId();
+        TaskQuery taskQuery = taskService.createTaskQuery().processInstanceId(processInstanceId);
+        taskQuery.or()
+            .taskCandidateOrAssigned(securityManager.getAuthenticatedUserId(),
+                securityManager.getAuthenticatedUserGroups())
+            .taskOwner(authenticatedUserId)
```

Comment:
who is the taskOwner in this case?

Reply:
Whoever is set to be the owner of the task. Setting the owner is only available via the API not possible via process definition.